File: go/pkg/models/image_generation.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type GenerationStatus int

const (
	GenerationStatusUnidentified GenerationStatus = iota
	GenerationStatusPending
	GenerationStatusInProgress
	GenerationStatusComplete
	GenerationStatusFailed
)

type GenerationBackend int

const (
	GenerationBackendUnidentified GenerationBackend = iota
	GenerationBackendImagine
)

type ImageGeneration struct {
	ID                  uuid.UUID         `json:"id"`
	CreatedAt           time.Time         `json:"createdAt"`
	UpdatedAt           time.Time         `json:"updatedAt"`
	UserID              uuid.UUID         `json:"userID"`
	GenerationID        string            `json:"generationID"`
	GenerationBackendID GenerationBackend `json:"generationBackendID"`
	Status              GenerationStatus  `json:"status"`
	OptionOne           *string           `json:"optionOne"`
	OptionTwo           *string           `json:"optionTwo"`
	OptionThree         *string           `json:"optionThree"`
	OptionFour          *string           `json:"optionFour"`
}

File: js/packages/types/src/imageGeneration.ts
export enum GenerationStatus {
  Unidentified,
  Pending,
  InProgress,
  GenerateImageOptions,
  Complete,
  Failed
}

export enum GenerationBackend {
  Unidentified,
  Imagine,
  UseApi
}

export type ImageGeneration = {
  id: string;
  createdAt: Date;
  updatedAt: Date;
  userId: string;
  generationId: string;
  generationBackendId: GenerationBackend;
  status: GenerationStatus;
  optionOne: string | null;
  optionTwo: string | null;
  optionThree: string | null;
  optionFour: string | null;
};
